src/library/scala/collection/immutable/StrictOptimizedSeqOps.scala

```diff
@@ -38,7 +37,7 @@ trait StrictOptimizedSeqOps[+A, +CC[_], +C]
       b += it.next()
       i += 1
     }
-    if (!it.hasNext) throw new IndexOutOfBoundsException(index.toString)
+    if (index < 0 || !it.hasNext) throw new IndexOutOfBoundsException(s"can't update at $index since bound is 0-$i")
```

Comment:
I don't follow this one. When `index < 0`, the upper bound would not be given by `$i`. I wonder if we couldn't keep the default `IOOB` message for negative indices, and leave the old check at the start of the method.

Reply:
You are right, `$i` wont be uppder bound when `index <0`.
My initial intent was to consolidate throwing `IOOB` into one place to remove redundancy.
However, it is NOT redundant.

- [ ] Split `index < 0` branch